src/com/amazon/corretto/crypto/provider/Loader.java

```diff
@@ -436,7 +433,7 @@ private static synchronized Path createPrivateTmpDir(final String prefix) throws
 
                 final Path privateDirFullPath = systemTempDir.resolve(privateTempDir.toString());
 
-                final Path result = Files.createDirectory(privateDirFullPath, SELF_OWNER_FILE_PERMISSIONS);
+                final Path result = Files.createDirectories(privateDirFullPath, SELF_OWNER_FILE_PERMISSIONS);
```

Comment:
should we catch IOException on L443 to retry directory creation on failure?

Reply:
The old code did not do that. Your suggestion is good and I'll add it to the PR.